weld/llvm.rs

```diff
@@ -806,53 +837,98 @@ impl LlvmGenerator {
                                  par_for: &ParallelForData,
                                  func: &SirFunction,
                                  ctx: &mut FunctionContext) -> WeldResult<()> {
+
         if !(par_for.data.len() == 1 && par_for.data[0].start.is_none()) {
             for iter in par_for.data.iter() {
-                let (data_ll_ty, data_ll_sym) = self.llvm_type_and_name(func, &iter.data)?;
-                let data_prefix = llvm_prefix(&data_ll_ty);
+                if iter.kind == IterKind::NdIter {
+                    /* Note: here we can't use num_iters_str as a proxy for end as the array may not be
+                     * contiguous */
+                    let data_llvm_info = self.get_array_llvm_info(func, ctx, &iter.data, "".to_string(), false)?;
+                    /* int offset = 0;
+                     * for (i = 0; i < len(shapes); i++) {
+                     *     int max_i = shapes[i] - 1;
+                     *     int stride_i = strides[i];
+                     *     offset += max_i*stride_i;
+                     * }
+                     * int max_val = start + offset;
+                     * cmp max_val, data_size_ll_tmp
+                     */
```

Comment:
This looks like dead code, but it's not? :)
Add a short comment above saying you're generating code for something similar to the following C code

Reply:
yup updated the comment above!